server/database.js

```diff
@@ -0,0 +1,35 @@
+const mysql = require('mysql');
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'root',
+  database: 'product_db',
+});
+
+const getProduct = function getProductInformation(productId, callback) {
+  connection.query(`select * from products where id=${productId}`, (err, results) => {
+    if (err) console.error(err);
+    callback(results);
+  });
+};
+
+const getRelated = function getRelatedProducts(productName, productId, callback) {
+  connection.query(`select id, product_tier, price, stock_count, thumbnail from products where name='${productName}' and id <> ${productId}`, (err, results) => {
+    if (err) console.error(err);
+    callback(results);
+  });
+};
+
+const getAll = function getProductAndRelatedProducts(productId, callback) {
+  let storage;
+  getProduct(productId, (results) => {
+    storage = results;
+    const productName = storage[0].name;
+    getRelated(productName, productId, (relatedResults) => {
+      storage = storage.concat(relatedResults);
+      callback(storage);
+    });
```

Comment:
Great function!  To clarify, the database is set up so that there are multiple products with same names but different ids? (i.e. Same product but different colors that you can toggle through)

Reply:
Yes, that's more or less how I've set it up. I believe I could refractor so the products have different colors, rather than different "tiers", which is how I've set it up now.

Reply:
Awesome, everything looks great so far!